### ISSUE TITLE
Add trait: Privacy-preserving crypto

### ISSUE THREAD

**Issue by jceb** [decentralized-identity/did-traits#16]
As pointed out in https://github.com/decentralized-identity/did-methods/issues/10#issuecomment-2500822445, I think this is a good trait to integrate.

**Comment by jceb** [comment]
Example definition:

```json
{
    "cryptography_privacy_preserving": {
      "type": "boolean",
      "title": "Privacy Preserving Cryptography",
      "description": "A DID method's ability to use cryptographic techniques that enable identity verification and authentication while minimizing the disclosure of sensitive personal information. Example: BBS."
    }
}
```

**Comment by ottomorac** [comment]
in addition to BBS in the example. Suggest we also mention Baby Jub Jub Cryptography (we use it in the did:iden3 mehtod based the open source Iden3 stack).  This is an EDDSA curve used for zk-snarks.

You can see the formal paper describing it here for reference:
https://upcommons.upc.edu/bitstream/handle/2117/361741/mathematics-09-03022.pdf?sequence=1

**Comment by jceb** [comment]
@ottomorac  I'm not familiar with this kind of crypto and just added BBS as an example. Could the algorithms be summarized under "zk-snarks"?

**Comment by ottomorac** [comment]
Yes let's keep it generic then without a specific crypto suite like BBS or BJJ. We could say instead "Zk-snarks, zk-starks or other similar zero knowledge proof techniques".

**Comment by jceb** [comment]
The trait has been added. It's up to the method author to provide detailed information on the supported techniques.
